### PR TITLE
"[ENH]Implement programmatic lookup for MTYPE and SCITYPE"

### PR DESCRIPTION
**Pull request by sun-9545sunoj**
Reference Issues/PRs
**Closes #9704**

What does this implement/fix? Explain your changes.

This PR implements a programmatic lookup for MTYPE and SCITYPE registries to resolve long-standing circular dependency issues between _registry.py and _check.py.

**Dynamic Attributes:** Replaced static dictionary/list definitions with PEP 562 __getattr__ logic to allow module-level attributes to be generated on-demand.

**Recursion Guard:** Implemented a global _IS_GENERATING reentrancy lock to prevent infinite recursion loops during attribute lookup, ensuring compatibility with IDE autocompletion and inspection tools.

**Metadata Extraction:** Updated the tabular registry (MTYPE_REGISTER_TABLE) to 4-element tuples to support lazy extraction of python_dependencies without triggering ModuleNotFoundError for missing soft dependencies like polars.

Contributor Update: Added myself to the .all-contributorsrc file with code and bug badges.

Does your contribution introduce a new dependency? If yes, which one?
**No.**

What should a reviewer concentrate their feedback on?

**The robustness of the _IS_GENERATING guard in handling edge cases during registry assembly.**

**The expansion of the MTYPE_REGISTER_TABLE tuples to ensure they correctly map to the new dynamic parsing logic.**

Did you add any tests for the change?
I verified the changes using the existing datatype test suite. **Run pytest skpro/datatypes/tests/ -v showing 107 passed tests. The logic successfully handles environments with and without optional soft dependencies.**

Any other comments?
**This cleanup provides a more maintainable foundation for the datatypes module and simplifies the internal import structure of the package.**

### CODE REVIEW

File: skpro/datatypes/_registry.py [hunk 1]
```diff
@@ -39,38 +40,127 @@
...
+                for mtype_tuple in MTYPE_REGISTER:
+                    mtype = mtype_tuple[0]
+                    scitype = mtype_tuple[1]
+                    scitypes.add(scitype)
+                    if len(mtype_tuple) >= 4 and mtype_tuple[3] is not None:
+                        deps = mtype_tuple[3]
+                        soft_deps[mtype] = (
+                            list(deps) if isinstance(deps, tuple) else deps
+                        )
+
```
[the remark this answers is not in the document]
Comment (sun-9545sunoj) [reply skpro/datatypes/_registry.py:114]: @copilot apply changes based on [this feedback](https://github.com/sktime/skpro/pull/986#discussion_r2979395724)

**Comment by sun-9545sunoj** [comment]
@copilot apply changes based on the comments in [this thread](https://github.com/sktime/skpro/pull/986#pullrequestreview-4025913467)
